### ISSUE TITLE
SumIf with empty string

### ISSUE THREAD

**Issue by Aurelianus** [EPPlusSoftware/EPPlus#624]
Hi,
the sumif function with criteria empty string should sum values on string.empty and `null`.

EPPlusTest.FormulaParsing.Excel.Functions.Math.SumIfEqualToEmptyString
```
[TestMethod]
public void SumIfEqualToEmptyString()
{
    _worksheet.Cells["A1"].Value = null;
    _worksheet.Cells["A2"].Value = string.Empty;
    _worksheet.Cells["A3"].Value = "Not Empty";
    _worksheet.Cells["B1"].Value = 1d;
    _worksheet.Cells["B2"].Value = 3d;
    _worksheet.Cells["B3"].Value = 5d;
    var func = new SumIf();
    IRangeInfo range1 = _provider.GetRange(_worksheet.Name, 1, 1, 3, 1);
    IRangeInfo range2 = _provider.GetRange(_worksheet.Name, 1, 2, 3, 2);
    var args = FunctionsHelper.CreateArgs(range1, "", range2);
    var result = func.Execute(args, _parsingContext);
    Assert.AreEqual(1d, result.Result);
}
```
The result should be B1+B2 = 4. Microsoft Excel would return this with formula `SUMIF(A1:A3,\"\",B1:B3)`. 

The intension to sum all empty cells could use the `IsBlank` function which in turn returns true for `null` and `string.Empty` 
see EPPlusTest.Excel.Functions
IsBlankShouldReturnTrueIfFirstArgIsNull
IsBlankShouldReturnTrueIfFirstArgIsEmptyString

Could you change the behavior?

**Comment by swmal** [comment]
Hi,
Thanks for reporting, we'll have a look at this.

**Comment by swmal** [comment]
I tested this against the current versions of Excel and is it like you say, that an empty string criteria should evaluate as a match against an empty string value in a cell. I have fixed this and it will be included in the next version. You can test it via our development nuget feed, see https://github.com/EPPlusSoftware/EPPlus/wiki/Using-our-Develop-Nuget-Feed

**Comment by Aurelianus** [comment]
Perfect, thank you.

**Comment by swmal** [comment]
This is included in the latest release on Nuget, v 5.8.9

**Comment by Aurelianus** [comment]
Hi,
it looks like that the fix is not complete.

1) [EPPlusTest.FormulaParsing.Excel.Functions.Math.SumIfTests].SumIfEqualToEmptyString fails if the formula has to be parsed
2) As synoymous syntax for selection empty cells exists with operator "=" as opposite to "<>" (unequal thus unempty). See Formlua B4 (works in Ms Excel)

`
[TestMethod]
        public void SumIfEqualToEmptyString_Parser()
        {
            using (var pck = new ExcelPackage())
            {
                var sheet = pck.Workbook.Worksheets.Add("test");
                _worksheet.Cells["A1"].Value = null;
                _worksheet.Cells["A2"].Value = string.Empty;
                _worksheet.Cells["A3"].Value = "Not Empty";
                _worksheet.Cells["B1"].Value = 1d;
                _worksheet.Cells["B2"].Value = 3d;
                _worksheet.Cells["B3"].Value = 5d;

                sheet.Cells["B4"].Formula = "SUMIF(A1:A3,\"=\",B1:B3)";
                sheet.Cells["B5"].Formula = "SUMIF(A1:A3,\"\",B1:B3)";
                sheet.Calculate();
                Assert.AreEqual(4d, sheet.Cells["B4"].Value);
                Assert.AreEqual(sheet.Cells["B4"].Value, sheet.Cells["B5"].Value);
            }
        }
`

3) Same applies for [EPPlusTest.FormulaParsing.Excel.Functions.Math.SumIfTests].SumIfNotEqualToNull
Following test fails

`
[TestMethod]
        public void SumIfNotEqualToNull_Parser()
        {
            using (var pck = new ExcelPackage())
            {
                var sheet = pck.Workbook.Worksheets.Add("test");
                _worksheet.Cells["A1"].Value = null;
                _worksheet.Cells["A2"].Value = string.Empty;
                _worksheet.Cells["A3"].Value = "Not Empty";
                _worksheet.Cells["B1"].Value = 1d;
                _worksheet.Cells["B2"].Value = 3d;
                _worksheet.Cells["B3"].Value = 5d;

                _worksheet.Cells["B4"].Formula = "SUMIF(A1:A3,\"<>\",B1:B3)";
                _worksheet.Calculate();
                Assert.AreEqual(8d, sheet.Cells["B4"].Value);
            }
        }
`
